internal/sockstate/netstat_linux.go

```
@@ -0,0 +1,208 @@
+	v, err := strconv.ParseUint(s, 16, 32)
+	if err != nil {
+		return nil, err
+	}
+	ip := make(net.IP, net.IPv4len)
+	binary.LittleEndian.PutUint32(ip, uint32(v))
+	return ip, nil
+}
+
+func parseAddr(s string) (*sockAddr, error) {
+	fields := strings.Split(s, ":")
+	if len(fields) < 2 {
+		return nil, fmt.Errorf("sockstate: not enough fields: %v", s)
+	}
+	var ip net.IP
+	var err error
+	switch len(fields[0]) {
+	case ipv4StrLen:
+		ip, err = parseIPv4(fields[0])
+	default:
+		log.Fatal("Bad formatted string")
+	}
+	if err != nil {
+		return nil, err
+	}
+	v, err := strconv.ParseUint(fields[1], 16, 16)
+	if err != nil {
+		return nil, err
+	}
+	return &sockAddr{IP: ip, Port: uint16(v)}, nil
+}
+
+func parseSocktab(r io.Reader, accept AcceptFn) ([]sockTabEntry, error) {
+	br := bufio.NewScanner(r)
+	tab := make([]sockTabEntry, 0, 4)
+
+	// Discard title
+	br.Scan()
+
+	for br.Scan() {
+		var e sockTabEntry
+		line := br.Text()
+		// Skip comments
+		if i := strings.Index(line, "#"); i >= 0 {
+			line = line[:i]
+		}
+		fields := strings.Fields(line)
+		if len(fields) < 12 {
+			return nil, fmt.Errorf("sockstate: not enough fields: %v, %v", len(fields), fields)
+		}
+		addr, err := parseAddr(fields[1])
+		if err != nil {
+			return nil, err
+		}
+		e.LocalAddr = addr
+		addr, err = parseAddr(fields[2])
+		if err != nil {
+			return nil, err
+		}
+		e.RemoteAddr = addr
+		u, err := strconv.ParseUint(fields[3], 16, 8)
+		if err != nil {
+			return nil, err
+		}
+		e.State = skState(u)
+		u, err = strconv.ParseUint(fields[7], 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		e.UID = uint32(u)
+		e.Ino = fields[9]
+		if accept(&e) {
+			tab = append(tab, e)
+		}
+	}
+	return tab, br.Err()
+}
+
+// tcpSocks returns a slice of active TCP sockets containing only those
+// elements that satisfy the accept function
+func tcpSocks(accept AcceptFn) ([]sockTabEntry, error) {
+	f, err := os.Open(pathTCPTab)
+	defer func() {
+		_ = f.Close()
```

if you're ignoring the error of f.Close() then just `defer f.Close()`

---

It's so GolangCI shut ups, which it doesn't which just a `defer close()`